src/runtime_src/hip/core/event.cpp

```diff
@@ -172,13 +172,15 @@ copy_buffer::copy_buffer(std::shared_ptr<stream> s)
 
 bool copy_buffer::submit()
 {
-  //handle = std::async(&xrt::bo::sync, &cbo, cdirection);
+  handle = std::async([this] { this->cbo.sync(this->cdirection); });
+  set_state(state::running);
   return true;
 }
 
 bool copy_buffer::wait()
 {
   handle.wait();
```

Comment:
This is cool, but to above comment, running may be deferred until exactly this point unless std::launch::async is speciied when calling std::async.